src/download_and_link.rs

```diff
@@ -65,6 +65,30 @@ impl DownloadAndLinkHooks for () {
 	type Error = Infallible;
 }
 
+/// Options for which dependencies to install.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum InstallDependenciesMode {
+	/// Install all dependencies.
+	All,
+	/// Install only dependencies, not dev_dependencies.
+	Prod,
+	/// Install only dev_dependencies.
+	Dev,
+}
+impl InstallDependenciesMode {
+	fn fits(&self, dep_ty: &DependencyType) -> bool {
```

Comment:
```suggestion

impl InstallDependenciesMode {
	fn fits(&self, dep_ty: DependencyType) -> bool {
```
Since `DependencyType` has the same size as a pointer let's just take ownership.